src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -228,7 +228,90 @@ def test_WINNF_FT_S_HBT_4(self):
+
+    # Register the device
+    device_a = json.load(open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_a['userId']})
+    request = {'registrationRequest': [device_a]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+
+    # Check registration response
+    self.assertEqual(response['response']['responseCode'], 0)
+    cbsd_id = response['cbsdId']
+    del request, response
+
+    # Request grant
+    grant_0 = json.load(open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_id
+    request = {'grantRequest': [grant_0]}
+
+    # Check grant response
+    response = self._sas.Grant(request)['grantResponse'][0]
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['response']['responseCode'], 0)
+    grant_id = response['grantId']
+    grant_expire_time = datetime.strptime(response['grantExpireTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+    del request, response
+
+    # First successful Heartbeat
+    request = {
+        'heartbeatRequest': [{
+            'cbsdId': cbsd_id, 'grantId': grant_id, 'operationState': 'GRANTED'
+        }]
+    }
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+
+    # Check the heartbeat response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['grantId'], grant_id)
+    self.assertEqual(response['response']['responseCode'], 0)
+    transmit_expire_time = datetime.strptime(response['transmitExpireTime'],
+                                             '%Y-%m-%dT%H:%M:%SZ')
+    del request, response
+
+    # Relinquish the grant
+    request = {
+        'relinquishmentRequest': [{
+            'cbsdId': cbsd_id, 'grantId': grant_id
+        }]
+    }
+    response = self._sas.Relinquishment(request)['relinquishmentResponse'][0]
+
+    # Check the relinquishment response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['grantId'], grant_id)
+    self.assertEqual(response['response']['responseCode'], 0)
+
+    # use relinquished grantId in new heartbeat request after transmitExpireTime
+    # is passed, but before the grant expires
+    request = {
+        'heartbeatRequest': [{
+            'cbsdId': cbsd_id, 'grantId': grant_id, 'operationState': 'GRANTED'
+        }]
+    }
+    transmit_expire_wait_time = (transmit_expire_time - datetime.utcnow()).total_seconds()
+    time.sleep(transmit_expire_wait_time + 1)
+    self.assertGreater(datetime.utcnow(), transmit_expire_time)
+    self.assertLess(datetime.utcnow(), grant_expire_time)
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
 
+    # Check the heartbeat response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertTrue(response['response']['responseCode'] in (103, 500))
+    self.assertLessEqual(
```

Comment:
Line 311 doesn't quite match the spec which reads: ""transmitExpireTime shall be no later than the current time."